homestar-runtime/tests/network/gossip.rs

```diff
@@ -0,0 +1,87 @@
+use crate::utils::{check_lines_for, kill_homestar, retrieve_output, stop_homestar, BIN_NAME};
+use anyhow::Result;
+use once_cell::sync::Lazy;
+use serial_test::file_serial;
+use std::{
+    fs,
+    path::PathBuf,
+    process::{Command, Stdio},
+    thread,
+    time::Duration,
+};
+
+static BIN: Lazy<PathBuf> = Lazy::new(|| assert_cmd::cargo::cargo_bin(BIN_NAME));
+
+#[test]
+#[file_serial]
+fn test_libp2p_receipt_gossip_serial() -> Result<()> {
+    let _ = stop_homestar();
+
+    let homestar_proc1 = Command::new(BIN.as_os_str())
+        .env(
+            "RUST_LOG",
+            "homestar=debug,homestar_runtime=debug,libp2p=debug,libp2p_gossipsub::behaviour=debug",
+        )
+        .arg("start")
+        .arg("-c")
+        .arg("tests/fixtures/test_gossip1.toml")
+        .arg("--db")
+        .arg("homestar10.db")
+        .stdout(Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    let homestar_proc2 = Command::new(BIN.as_os_str())
+        .env(
+            "RUST_LOG",
+            "homestar=debug,homestar_runtime=debug,libp2p=debug,libp2p_gossipsub::behaviour=debug",
+        )
+        .arg("start")
+        .arg("-c")
+        .arg("tests/fixtures/test_gossip2.toml")
+        .arg("--db")
+        .arg("homestar11.db")
+        .stdout(Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    // Wait for nodes to be come online
+    // TODO Wait for connect events over websocket
+    thread::sleep(Duration::from_secs(2));
```

Comment:
instead of this, let's use the tcp stream retry trick in other tests.

Reply:
Done. 👌